stingray/modeling/gpmodeling.py

```diff
@@ -485,20 +512,41 @@ def get_prior(params_list, prior_dict):
 
     def prior_model():
         prior_list = []
-        for i in params_list:
-            if isinstance(prior_dict[i], tfpd.Distribution):
-                parameter = yield Prior(prior_dict[i], name=i)
-            elif isinstance(prior_dict[i], Prior):
-                parameter = yield prior_dict[i]
+        for param in params_list:
+            if param == "log_alpha":
+                for j in range(p):
```

Comment:
Is log_alpha here a single parameter, or is it a list or array of parameters, if so what is its size?

Reply:
`log_alpha` is an array of size `p`, `log_beta` an array of size `q`

Reply:
This method seems fine, but some suggestions:
1) Here all the `log_alpha`'s will have the same prior distribution. Is this intended?
2) Instead of checking for `params == "log_alpha"` at each instance, we may want to just check if the parameter the user is giving is an array and do its part separately. For example usually the dictionary we insert to the `get_prior` function is like  
```python
prior_dict = {
   "log_alpha": tfpd.Uniform(low = jnp.log(1e-1), high = jnp.log(2e+2)),
   "t0": tfpd.Uniform(low = 0.0 - 0.1, high = 1 + 0.1),
   }
```
If we want `log_alpha`, to be actually be an array parameter  of size p, we can change the dictionary to 
```python
prior_dict = {
   "log_alpha": [tfpd.Uniform(low = jnp.log(1e-1), high = jnp.log(2e+2)),p],
   "t0": tfpd.Uniform(low = 0.0 - 0.1, high = 1 + 0.1),
   }
```
and the corresponding modifications to the `get_prior` function's `prior_model` could be:
```python
def prior_model():
   prior_list = []
   for i in params_list:
      # New code added for array parameters
      if isinstance(prior_dict[i], list):
         for j in range(prior_dict[i][1]):
            if instance(prior_dict[param][0], tfpd.Distribution):
               parameter = yield Prior(prior_dict[param][0], name = param + str(j))
            elif isinstance(prior_dict[param][0], Prior):
               parameter = yield prior_dict[param]
            else:
               raise ValueError("Prior should be a tfpd distribution or a jaxns prior.")
         prior_list.append(parameter)
      # Old code
      elif isinstance(prior_dict[i], tfpd.Distribution):
         parameter = yield Prior(prior_dict[i], name=i)
         prior_list.append(parameter)
      elif isinstance(prior_dict[i], Prior):
         parameter = yield prior_dict[i]
         prior_list.append(parameter)
      else:
         raise ValueError("Prior should be a tfpd distribution or a jaxns prior.")
   return tuple(prior_list)
```